components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthRequestValidator.java

```diff
@@ -173,21 +183,30 @@ private void valiateTransactionContext(JWTClaimsSet claimsSet) throws CibaAuthFa
     private void validateBindingMessage(JWTClaimsSet claimsSet) throws CibaAuthFailureException {
 
         try {
-            // Validation for binding_message.
-            if ((claimsSet.getClaim(CibaConstants.BINDING_MESSAGE)) == null) {
-                // Request has claim for binding_message.
-                return;
-            }
 
-            if (StringUtils.isBlank(claimsSet.getStringClaim(CibaConstants.BINDING_MESSAGE))) {
-                // Binding_message with a blank value which is not acceptable.
-                if (log.isDebugEnabled()) {
-                    log.debug("Invalid CIBA Authentication Request made by client with clientID : " +
-                            claimsSet.getIssuer() +
-                            ".The request is with invalid  value for (binding_message).");
+            // Request has claim for binding_message(optional).
+            if (claimsSet.getClaim(CibaConstants.BINDING_MESSAGE) != null) {
+                // Validate binding message is not blank
+                String bindingMessage = claimsSet.getStringClaim(CibaConstants.BINDING_MESSAGE);
+                if (StringUtils.isBlank(bindingMessage)) {
+                    // Binding_message with a blank value which is not acceptable.
+                    if (log.isDebugEnabled()) {
+                        log.debug("Invalid CIBA Authentication Request made by client with clientID : " +
+                                claimsSet.getIssuer() +
+                                ".The request is with invalid  value for (binding_message).");
+                    }
+                    throw new CibaAuthFailureException(ErrorCodes.INVALID_BINDING_MESSAGE,
+                            "Invalid value for (binding_message).");
+                } else {
+                    // Validate binding message contain only plain text
+                    Pattern specialCharactersPattern = Pattern.compile("[^a-z0-9 ]", Pattern.CASE_INSENSITIVE);
+                    Matcher specialCharactersPatternMatcher = specialCharactersPattern.matcher(bindingMessage);
+                    boolean isSpecialCharactersAvailable = specialCharactersPatternMatcher.find();
```

Comment:
Is there a simpler way to check this?
boolean isAlphaNumeric = s != null && s.chars().allMatch(Character::isLetterOrDigit);

Reply:
What would be the max size/length allowed of a binding message?

Reply:
Re size, length the specification doesn't specify anything apart from mentioning A human-readable identifier or message, SHOULD be relatively short and use a limited set of plain text characters. Refer https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#auth_request

Reply:
For the moment we left the size open-ended. Any suggestions?

Reply:
It is a pass through IIRC and we don't persist IIRC. @KalanaDananjaya correct me if I am wrong.

Reply:
+1 for the new method. Replaced with  boolean isAlphaNumeric = s.chars().allMatch(Character::isLetterOrDigit); as binding message is already checked for null above